packages/migrate/test/migrate.test.ts

```diff
@@ -0,0 +1,430 @@
+    let reporter: Reporter;
+
+    beforeEach(async () => {
+      project = Project.fromDir(projectPath, { linkDeps: true, linkDevDeps: true });
+
+      delete project.files['index.js'];
+
+      await project.write();
+
+      logger = createLogger({
+        transports: [new transports.Console({ format: format.cli(), level: 'debug' })],
+      });
+
+      reporter = new Reporter(
+        {
+          tsVersion: '',
+          projectName: '@rehearsal/test',
+          basePath: project.baseDir,
+          commandName: '@rehearsal/migrate',
+        },
+        logger
+      );
+    });
+
+    afterEach(() => {
+      project.dispose();
+    });
+
+    test('it works', async () => {
+      const [inputs, outputs] = prepareInputFiles(project, [
+        'missing-local-prop.hbs',
+        'missing-local-prop.js',
+      ]);
+
+      const input: MigrateInput = {
+        basePath: project.baseDir,
+        sourceFiles: inputs,
+        entrypoint: '',
+        reporter,
+        logger,
+      };
+
+      await migrate(input);
+
+      const expectedTs = `import Component from '@glimmer/component';
+
+export default class Foo extends Component {}
+`;
+
+      const expectedHbs = `{{! @glint-expect-error @rehearsal TODO TS2339: Property 'name' does not exist on type 'Foo'. }}
+{{! @glint-expect-error @rehearsal TODO TS2339: Property 'age' does not exist on type '{}'. }}
+<span>Hello, I am {{this.name}} and I am {{@age}} years old!</span>
+`;
+
+      expectFile(outputs[0]).toEqual(expectedHbs);
+      expectFile(outputs[1]).toEqual(expectedTs);
+    });
+  });
+
+  describe('.js', () => {
+    let project: Project;
+    let logger: Logger;
+    let reporter: Reporter;
+
+    beforeEach(async () => {
+      project = Project.fromDir(projectPath, { linkDeps: true, linkDevDeps: true });
+
+      logger = createLogger({
+        transports: [new transports.Console({ format: format.cli(), level: 'debug' })],
+      });
+
+      reporter = new Reporter(
+        {
+          tsVersion: '',
+          projectName: '@rehearsal/test',
+          basePath: project.baseDir,
+          commandName: '@rehearsal/migrate',
+        },
+        logger
+      );
+    });
+
+    test('class with missing prop', async () => {
+      project.mergeFiles({
+        src: {
+          'foo.js': `class Foo {
+  hello() {
+    return this.name;
+  }
+}
+`,
+        },
+      });
+
+      await project.write();
+
+      const [inputs, outputs] = prepareInputFiles(project, ['foo.js']);
+
+      const input: MigrateInput = {
+        basePath: project.baseDir,
+        sourceFiles: inputs,
+        entrypoint: '',
+        reporter,
+        logger,
+      };
+
+      await migrate(input);
+
+      const expected = `class Foo {
+  hello() {
+/* @ts-expect-error @rehearsal TODO TS2339: Property 'name' does not exist on type 'Foo'. */
+    return this.name;
+  }
+}
+`;
+
+      expectFile(outputs[0]).toEqual(expected);
+    });
+
+    test.skip('complex', async () => {
```

Comment:
skip?

Reply:
I looked into this on Saturday. This was the test skipped in the ESM migration. The issue is the way this test is architected. UUID and importer types are being used elsewhere in the monorepo and the types are already available. As such the updated package.json devDep will never be included (since it already is). 

Additionally in debugging (can’t 100% validate but is replicated) if we specify a different module which doesn’t bundle types and we don’t already consume it. TypeScript 4.9 is auto pulling in the correlated @types in the typescript global cache.